INN/umbrella-currentorg
# 137 filtering hs

## Changes

This pull request implements the following portion of https://github.com/INN/umbrella-currentorg/issues/137#issuecomment-639686424:

- submit the form when the `<select>` changes
- track whether the inputs in the details element are changed, and how many are checked
- when the <details> element is closed or otherwise onBlur, if the inputs there are changed, submit the search
- limit the number of checked inputs to no more than 3: if it's 3, disable all inputs until one is unchecked

It also:

- makes the `summary` in the `details` full width when the `details` is open

![Screen Shot 2020-06-05 at 18 23 41](https://user-images.githubusercontent.com/1754187/83927076-c1964480-a759-11ea-87d7-7866061a73b6.png)


## Why



For https://github.com/INN/umbrella-currentorg/issues/137

## Testing/Questions

Features that this PR affects:

- search form


Questions that need to be answered before merging:


Steps to test this PR:

1. check that the form submits when the select's value changes
2. Check that the form submits when the `<details>` closes, only if the checkboxes checked have changed.
    - uncheck and check the same box: should not submit
    - check a new box: should submit
    - uncheck a checked box: should submit
3. check that if three checkboxes are checked, other checkboxes in that dropdown are `disabled`:
    - on page load, with three checked
    - after checking three
